test/AutoDiff/autodiff_indirect_diagnostics.swift

```diff
@@ -6,14 +6,13 @@
 
```

Comment:
I'm actually not sure what's the expected behavior for these tests. I'll debug a bit to see what's going on.

At a glance, the changes seem unideal. The old diagnostics were more informative and appeared on the `gradient` user invocation.
```diff
--- a/test/AutoDiff/autodiff_indirect_diagnostics.swift
+++ b/test/AutoDiff/autodiff_indirect_diagnostics.swift
@@ -6,14 +6,13 @@

 // Test unmet generic requirements.

-// expected-error @+1 {{function is not differentiable}}
 @differentiable
-// expected-note @+1 {{when differentiating this function definition}}
 func generic<T: Differentiable & FloatingPoint>(_ x: T) -> T {
+  // expected-error @+2 {{expression is not differentiable}}
   // expected-note @+1 {{member is not differentiable because the corresponding protocol requirement is not '@differentiable'}}
   return x + 1
 }
-_ = gradient(at: 1.0, in: generic) // expected-error {{function is not differentiable}}
+_ = gradient(at: 1.0, in: generic)
```

Reply:
I didn't quite make sense of the diagnostic behavior changes yet.
We plan to change/fix diagnostic behavior soon, so it would be good to revisit these changes as part of that follow-up.